### ISSUE TITLE
Data lost when changing file name

### ISSUE THREAD

**Issue by gacekssj4** [airsonic/airsonic#1190]
After looking into Issues I found one I wanted: #416 

I understand approach of:
- Do not modify - since it can be damaged (can be left as option tho)
- Storage of multiple users data etc

However, for me, its very inconvenient that renaming any kind of folder/file name leads to meta data loss.

If I may suggest 2 approaches to the case:

### First (Intrusive)
After getting "ID" in database of SubSonic - Store it in some kind of special subsonic meta tag.
So changing folder structure renamed entries can be paired back to their database entries.

Would still allow to store multiple reviews per user and any change to folder/file structure would not lead to loss of data... (sorry, I've been working on this data!)

### Second (Unintrusive)
Create special file, for example "subsonic.json" which contains filenames (and or some other file identifying data) which also stores ID from Subsonic database. If file exists, try to rebuild relations.

Would help with 2 above.

**Comment by jvoisin** [comment]
My airsonic instance has read-only access to my files, so the first solution doesn't work.

Your second solution doesn't scale, and also won't work: if you move a file, how will subsonic know that it moved?

Also, I don't think that airsonic is storing any media-specific metadata: you just have to trigger a re-scan for your file to be present again.

**Comment by gacekssj4** [comment]
Second solution is more for "album" moving and categorizing. And also requires rw.
I think it's based on "filename". Maybe at last changing it to "Artist/Album/Trackname" as option.

**Comment by gacekssj4** [comment]
PS: I Forgot to add, that airsonic.json would be stored inside of Album folder. Making it stick to album.

**Comment by eharris** [comment]
I would propose another solution, which would be to use presumably unique-enough identifiers that are already stored in the individual files in order to track and propagate the database info when files are moved.  This would require both changes to support and store additional tags (see related issues #164, #717) as well as changes to the scanning service to support identifying a "new" file as being the same one that is missing based on the unique tag value.

For instance, MusicBrainz tags `musicbrainz_recordingid` and `acoustid_id` would probably be pretty good candidates to be used for this.

**Comment by gacekssj4** [comment]
Anything that would allow me (or other people who want) to keep metadata persistent, cause using AirSonic would render me unable to improve my music library... 

Looking at MusicBrainz its really great idea as it would also allow to score one track and duplicate would be rated too!

**Comment by tesshucom** [comment]
Since there are cases where multiple identical songs exist below vaious, it is impossible to confuse file structure with id3.
On API design, Some properties (star, etc.) exist in both file structure and id3.
The initial design may have taken into account how to handle them well.

But in the end, it was difficult to guarantee such a clever mechanism locally, and it was nonstandard compared to many software specifications.
Even if it can be realized with a Web interface, existing client software will not follow.

For this reason, provisional information such as Star and Count has been able to transfer to external services via REST.
Data retention with external services avoids some of the difficult problems such as erasing Server data and replaying from different platforms.

I think that there is such a history of development.

I think jvoisin's first comment is quite reasonable.

**Comment by gacekssj4** [comment]
Hello, thank you for comment.

I think design of app is good and totally understand your point of view. Maybe it's just not for me, as I'm looking for app that will give me personal music service (to listen in car, on trip, in office in home etc).
Ofcourse, there are disadventages of storing everything in idv3 as you stated and one of most trobule some would be data persistance - in case of replacing music file I would still need to rate all them differently...

Tbh I'm in trouble here myself.. as you stated 3rd party API I think would be best.

After reading all your comments and thinking. Storing data as "Artist/Album/Song" would be best solution.

I think that's what 3rd party serives do somehow. But I do not trust them... Just like making playlists on youtube - songs disappear - There are many more examples how external services can change... for example after getting alot of clients can change request quota etc. So basically they're all untrusty - at last for me. But that's not the topic here.

So basically I would like to see Airsonic as Metadata container (after reading your comments) but maybe with improved metadata assignment?

Thank you for all your thoughts - here is mine. Keep it in mind od discard it :) Best regards!

**Comment by tesshucom** [comment]
It may be off topic, but recently there seems to be something like this.

 - [Last.FM Compatible API for ListenBrainz](https://listenbrainz.readthedocs.io/en/latest/dev/api-compat.html)
 - [metabrainz/listenbrainz-server](https://github.com/metabrainz/listenbrainz-server)

Run Lastfm-compatible services in parallel with Docker and complete the same thing as Lastfm integration on your own local network.

If we can do that.
If it is reliable and the load is not a problem, it appears to be a modern solution.

(Or make such a microservice. . .? It's hard!)

There is also idea to add an extreme option to the Airsonic.
However, various other software is frustrated and it seems to cause the complexity of the main body.

**Comment by eharris** [comment]
Note that there is currently a PR #1224 to add support for ListenBrainz

**Comment by tesshucom** [comment]
I'll close it once. If you have any problems, please make another request again.
